fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -367,30 +396,141 @@ void parseKmsResponse(Reference<RESTKmsConnectorCtx> ctx,
+		                         expireAfterSec);
 	}
 
-	if (doc.HasMember(KMS_URLS_TAG)) {
-		try {
-			extractKmsUrls(ctx, doc, resp);
-		} catch (Error& e) {
-			TraceEvent("RefreshKmsUrls_Failed", ctx->uid).error(e);
-			// Given cipherKeyDetails extraction was done successfully, ignore KmsUrls parsing error
+	checkDocForNewKmsUrls(ctx, resp, doc);
+
+	return result;
+}
+
+Standalone<VectorRef<BlobMetadataDetailsRef>> parseBlobMetadataResponse(Reference<RESTKmsConnectorCtx> ctx,
+                                                                        Reference<HTTP::Response> resp) {
+	// Acceptable response payload json format:
+	// (baseLocation and partitions follow the same properties as described in BlobMetadataUtils.h)
+	//
+	// response_json_payload {
+	//   "blob_metadata_details" : [
+	//     {
+	//        "domain_id" : <domainId>,
+	//        "domain_name" : <baseCipher>,
+	//        "baseLocation" : <baseLocation>, (Optional if partitions is present)
+	//        "partitions" : [
+	//			  "partition1", "partition2", ...
+	//		  ], (Optional if baseLocation is present)
+	//        "refresh_after_sec"   : <refreshCipherTimeInterval>, (Optional)
+	//        "expire_after_sec"    : <expireCipherTimeInterval>  (Optional)
+	//     },
+	//     {
+	//         ....
+	//	   }
+	//   ],
+	//   "kms_urls" : [
+	//         "url1", "url2", ...
+	//   ],
+	//	 "error" : {					// Optional, populated by the KMS, if present, rest of payload is ignored.
+	//		"errMsg" : <message>,
+	//		"errCode": <code>
+	// 	  }
+	// }
+
+	if (resp->code != HTTP::HTTP_STATUS_CODE_OK) {
+		// STATUS_OK is gating factor for REST request success
+		throw http_request_failed();
+	}
+
+	rapidjson::Document doc;
+	doc.Parse(resp->content.c_str());
+
+	checkResponseForError(ctx, doc);
+
+	Standalone<VectorRef<BlobMetadataDetailsRef>> result;
+
+	// Extract CipherKeyDetails
+	if (!doc.HasMember(BLOB_METADATA_DETAILS_TAG) || !doc[BLOB_METADATA_DETAILS_TAG].IsArray()) {
+		TraceEvent(SevWarn, "ParseBlobMetadataResponse_FailureMissingDetails", ctx->uid).log();
+		throw operation_failed();
+	}
+
+	for (const auto& detail : doc[BLOB_METADATA_DETAILS_TAG].GetArray()) {
+		if (!detail.IsObject()) {
+			TraceEvent(SevWarn, "ParseBlobMetadataResponse_FailureDetailsNotObject", ctx->uid)
+			    .detail("Type", detail.GetType());
+			throw operation_failed();
+		}
+
+		const bool isDomainIdPresent = detail.HasMember(BLOB_METADATA_DOMAIN_ID_TAG);
+		const bool isDomainNamePresent = detail.HasMember(BLOB_METADATA_DOMAIN_NAME_TAG);
+		const bool isBasePresent = detail.HasMember(BLOB_METADATA_BASE_LOCATION_TAG);
+		const bool isPartitionsPresent = detail.HasMember(BLOB_METADATA_PARTITIONS_TAG);
+		if (!isDomainIdPresent || !isDomainNamePresent || (!isBasePresent && !isPartitionsPresent)) {
+			TraceEvent(SevWarn, "ParseBlobMetadataResponse_MalformedDetail", ctx->uid)
+			    .detail("DomainIdPresent", isDomainIdPresent)
+			    .detail("DomainNamePresent", isDomainNamePresent)
```

Comment:
`DomainName` might contains sensitive information; not sure if we want to log it?

Reply:
Agreed, but here this is just whether it's present, it doesn't actually log the domain name itself.